File: Library-Backend/src/main/java/com/system/Library_Backend/service/BookService.java
package com.system.Library_Backend.service;

import com.system.Library_Backend.model.Book;
import com.system.Library_Backend.repository.BookRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.Optional;

@Service
public class BookService {

    private final BookRepository bookRepository;

    @Autowired
    public BookService(BookRepository bookRepository) {
        this.bookRepository = bookRepository;
    }

    // Get all books
    public List<Book> getAllBooks() {
        return bookRepository.findAll();
    }

    // Get a single book by ID
    public Optional<Book> getBookById(Long id) {
        return bookRepository.findById(id);
    }

    // Add a new book
    public Book addBook(Book book) {
        return bookRepository.save(book);
    }

    // Update an existing book
    public Book updateBook(Long id, Book bookDetails) {
        Book book = bookRepository.findById(id)
                .orElseThrow(() -> new RuntimeException("Book not found with id: " + id));

        book.setTitle(bookDetails.getTitle());
        book.setAuthor(bookDetails.getAuthor());
        book.setAvailable(bookDetails.isAvailable());

        return bookRepository.save(book);
    }

    // Delete a book
    public void deleteBook(Long id) {
        Book book = bookRepository.findById(id)
                .orElseThrow(() -> new RuntimeException("Book not found with id: " + id));
        bookRepository.delete(book);
    }

    // Additional business logic methods can be added here
    // Example: Find all available books
    public List<Book> getAvailableBooks() {
        return bookRepository.findByAvailable(true);
    }
}

File: library-frontend/src/services/bookService.js
import API from './api';

export const getAllBooks = async () => {
  try {
    const response = await API.get('/books');
    return response.data;
  } catch (error) {
    console.error('Error fetching books:', error);
    throw error;
  }
};

export const createBook = async (bookData) => {
  try {
    const response = await API.post('/books', bookData);
    return response.data;
  } catch (error) {
    console.error('Error creating book:', error);
    throw error;
  }
};

export const updateBook = async (id, bookData) => {
  try {
    const response = await API.put(`/books/${id}`, bookData);
    return response.data;
  } catch (error) {
    console.error('Error updating book:', error);
    throw error;
  }
};

export const deleteBook = async (id) => {
  try {
    const response = await API.delete(`/books/${id}`);
    return response.data;
  } catch (error) {
    console.error('Error deleting book:', error);
    throw error;
  }
};



// // Book Service
// import api from './api'; // Adjust the import path as necessary
// export const bookService = {
//   getAllBooks: () => api.get('/books'),
//   getBookById: (id) => api.get(`/books/${id}`),
//   getAvailableBooks: () => api.get('/books/available'),
//   addBook: (book) => api.post('/books', book),
//   updateBook: (id, book) => api.put(`/books/${id}`, book),
//   deleteBook: (id) => api.delete(`/books/${id}`),
// };
